Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1401,6 +1401,17 @@ extension BrowserViewController: WKNavigationDelegate {
                     return
                 }
 
+                if let readerMode = tab.getHelper(name: ReaderMode.name()) as? ReaderMode {
+                    self.urlBar.updateReaderModeState(readerMode.state)
+                    if readerMode.state == .Active {
+                        self.showReaderModeBar(animated: false)
+                    } else {
+                        self.hideReaderModeBar(animated: false)
+                    }
+                } else {
+                    self.urlBar.updateReaderModeState(ReaderModeState.Unavailable)
```

Comment:
Note that we already update to the `Unavailable` state on `didStartProvisionalNavigation`, so we don't need these else condition do we?